### PR TITLE
Feature/mcw db 185604 update

### PR DESCRIPTION
**Pull request by xe21500**
Update dependencies & add common lifecycle stuff to the repo.

### CODE REVIEW

File: addon/storages/local.js [hunk 1]
```diff
@@ -27,6 +23,8 @@ export default ObjectProxy.extend(StorageMixin, {
 	 * @type {Object}
 	 * @default window.localStorage
 	 */
+	// "ember/no-arrow-function-computed-properties" conflicts with "prefer-arrow-callback"
+	// eslint-disable-next-line ember/no-arrow-function-computed-properties
 	adapter: computed(() => window.localStorage),
```
Comment (adrigzr) [inline addon/storages/local.js]:
```suggestion
	adapter: window.localStorage,
```

File: .huskyrc.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+	hooks: {
+		'commit-msg': 'commitlint -E HUSKY_GIT_PARAMS',
+		'pre-push': 'npm run lint:js && npm run lint:hbs && npm test'
```
Comment (adrigzr) [inline .huskyrc.js]:
```suggestion
		'pre-push': 'npm run lint && npm test'
```

File: addon/storages/memory.js [hunk 3]
```diff
@@ -23,7 +20,9 @@ export default ObjectProxy.extend(StorageMixin, {
 	 * @type {Object}
 	 * @default Ember.Object
 	 */
-	initialContent: computed(() => Ember.Object.create()),
+	// "ember/no-arrow-function-computed-properties" conflicts with "prefer-arrow-callback"
+	// eslint-disable-next-line ember/no-arrow-function-computed-properties
+	initialContent: computed(() => EmberObject.create()),
```
Comment (adrigzr) [inline addon/storages/memory.js]:
```suggestion
	init() {
	  this._super(...arguments);
	  this.initialContent = EmberObject.create();
	}
```

File: addon/storages/session.js [hunk 4]
```diff
@@ -20,6 +19,8 @@ export default LocalStorage.extend({
 	 * @type {Object}
 	 * @default window.localStorage
 	 */
+	// "ember/no-arrow-function-computed-properties" conflicts with "prefer-arrow-callback"
+	// eslint-disable-next-line ember/no-arrow-function-computed-properties
 	adapter: computed(() => window.sessionStorage),
```
Comment (adrigzr) [inline addon/storages/session.js]:
```suggestion
	adapter: window.sessionStorage,
```

File: package.json [hunk 5]
```diff
@@ -5,65 +5,86 @@
   "keywords": [
     "ember-addon"
   ],
+  "repository": "",
```
Comment (adrigzr) [inline package.json]:
```suggestion
  "repository": {
    "type": "git",
    "url": "https://github.com/BBVAEngineering/ember-storages.git"
  },
```

File: package.json [hunk 6]
```diff
@@ -5,65 +5,86 @@
...
+    "lint": "npm run lint:js && npm run lint:hbs",
+    "posttest": "npm run report-coverage && npm run check-coverage && npm run validate-coverage",
+    "report-coverage": "istanbul report --include=coverage/coverage-final.json text",
+    "semantic-release": "semantic-release",
+    "validate-coverage": "test \"$(cat coverage/coverage-summary.json | json total.lines.total)\" -gt 0"
   },
   "dependencies": {
-    "ember-cli-babel": "^6.6.0"
+    "ember-cli-babel": "^7.17.2",
+    "ember-cli-htmlbars": "^4.2.2"
```
Comment (adrigzr) [inline package.json]: Is this necessary?

**Review by adrigzr — CHANGES_REQUESTED** [review]

**Review by svsalvador — DISMISSED** [review]

File: addon/storages/local.js (hunk 1 above)
Comment (xe21500) [reply addon/storages/local.js]: Done

File: .huskyrc.js (hunk 2 above)
Comment (xe21500) [reply .huskyrc.js]: Done

File: addon/storages/memory.js (hunk 3 above)
Comment (xe21500) [reply addon/storages/memory.js]: I couldn't manage to make this one work, seems that no matter where I put the `_super`, there is no `initialContent` at some point during the tests and they break cathastropically. I'd say lets this one stay for now.

File: addon/storages/session.js (hunk 4 above)
Comment (xe21500) [reply addon/storages/session.js]: Done

File: package.json (hunk 6 above)
Comment (xe21500) [reply package.json]: Yep, it is

**Review by adrigzr — APPROVED** [review]

**Pull request merged**